src/magma/vk/textures/CustomImageView.cpp

```diff
@@ -0,0 +1,61 @@
+/**
+ * @file CustomImageView.cpp
+ * @brief Class contain custom image view
+ * @version 0.1
+ * @date 2021-02-13
+ */
+#include "magma/vk/textures/CustomImageView.h"
+
+#include "magma/vk/vulkan_common.h"
+#include <vulkan/vulkan_core.h>
+
+CustomImageView::CustomImageView(VkDevice device, Texture &texture, VkImageAspectFlags aspectMask) 
+        : _device(device) {
+    VkImageCreateInfo info = texture.getInfo()->imageInfo;
+    VkImageViewCreateInfo viewInfo{};
+    viewInfo.sType = VK_STRUCTURE_TYPE_IMAGE_VIEW_CREATE_INFO;
+    viewInfo.image = texture.getImage();
+    viewInfo.viewType = VK_IMAGE_VIEW_TYPE_2D;
+    viewInfo.format = info.format;
+    viewInfo.subresourceRange.aspectMask = aspectMask;
+    viewInfo.subresourceRange.baseMipLevel = 0;
+    viewInfo.subresourceRange.levelCount = 1;
+    viewInfo.subresourceRange.baseArrayLayer = 0;
+    viewInfo.subresourceRange.layerCount = 1;
+
+    VkImageView textureView;
+    VkResult result = vkCreateImageView(device, &viewInfo, nullptr, &_imageView);
+    VK_CHECK_ERR(result, "CustomImageView::failed to create image view!");
```

Comment:
nit: "::" looks a bit weird. I'd better change it to ":" + space

Reply:
Done